native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m

```diff
@@ -154,20 +142,19 @@ - (void)applicationDidBecomeActive:(UIApplication *)application
 {
     SFUserAccount *currentAccount = [SFUserAccountManager sharedInstance].currentUser;
     BOOL isAuthenticated = (!currentAccount.isTemporaryUser && !currentAccount.isAnonymousUser && currentAccount.credentials.refreshToken.length > 0);
-    if (!_jwtAuthenticating && !isAuthenticated) {
-        __weak __typeof(self) weakSelf = self;
-        [[SFAuthenticationManager sharedManager] loginWithCompletion:^(SFOAuthInfo * _Nonnull authInfo) {
-            __strong __typeof(self) strongSelf = weakSelf;
-            [strongSelf log:SFLogLevelInfo format:@"Authentication (%@) succeeded.  Launch completed.", authInfo.authTypeDescription];
-            [strongSelf setupRootViewController];
-        } failure:^(SFOAuthInfo * _Nonnull authInfo, NSError * _Nonnull authError) {
-            __strong __typeof(self) strongSelf = weakSelf;
-            [strongSelf log:SFLogLevelError format:@"Authentication (UserAgentFlow) failed: %@.", authError.localizedDescription];
-        }];
+    if (!isAuthenticated) {
+        if (!_jwtAuthenticating) {
+            [self startAuthentication:nil];
+        }
+    } else {
+        [self setupRootViewController];
     }
 }
 
-
+- (void)applicationDidEnterBackground:(UIApplication *)application
+{
+    [[SFAuthenticationManager sharedManager] cancelAuthentication];
```

Comment:
If user backgrounds the app, any in-progress authentication should be started over.